data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ExpressionEvaluator.java

```diff
@@ -17,7 +17,7 @@
      * @param context event used to resolve external references in the statement
      * @return coerced result of statement evaluation
      *
-     * @throws ClassCastException when unable to course statement result to type T
+     * @throws ExpressionEvaluationException if unable to evaluate or course the statement result to type T
```

Comment:
typo: You have "course", but I think you mean "cast" or "coerce." Do we coerce or only cast?